app/controllers/admin/payment_confirmation_report_uploads_controller.rb

```diff
@@ -0,0 +1,20 @@
+module Admin
+  class PaymentConfirmationReportUploadsController < BaseAdminController
+    before_action :ensure_service_operator
+
+    def new
+      @payroll_run = PayrollRun.find(params[:payroll_run_id])
+    end
+
+    def create
+      @payroll_run = PayrollRun.find(params[:payroll_run_id])
+      @payment_confirmation = PaymentConfirmation.new(@payroll_run, params[:file])
+      if @payment_confirmation.ingest
+        @payroll_run.update!(confirmation_report_uploaded_by: admin_session.user_id)
+        redirect_to admin_payroll_runs_path, notice: "Payment Confirmation Report successfully uploaded"
+      else
+        render :new, status: :bad_request
```

Comment:
I thought that a 200 response to a POST was to be interpreted as "entity created". Here's what the spec has to say:

<blockquote>
200 OK: The request has succeeded. The information returned with the response is dependent on the method used in the request, for example:

...

POST an entity describing or containing the result of the action; 
</blockquote>

And so I thought that we could use a 400 Bad Request to say "nothing has been created because the request contained invalid data". Does this not align with common usage in web apps?

Reply:
Interesting! When I read "POST an entity describing or containing the result of the action;", I don't think that necessarily translates as "entity created". My reading is that I should see "the result of the action", which may or may not be that the thing was created. In either case, at least for HTML-based requests, the standard pattern for RESTful Rails controllers is to re-render the form without a custom error code (see what Rails generates as a scaffold: https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/scaffold_controller/templates/controller.rb.tt)